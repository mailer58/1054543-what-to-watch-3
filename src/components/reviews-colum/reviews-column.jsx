import React from 'react';
import PropTypes from "prop-types";
import moment from 'moment';

export const ReviewsColumn = ({halfReviews}) => {
  return (
    <div className="movie-card__reviews-col">
      {halfReviews.map((review) => {
        let date = moment(review.date);
        const dateTime = date.format(`YYYY-MM-DD`);
        date = date.format(`MMM DD, YYYY`);

        return (
          <div key={review.id} className="review">
            <blockquote className="review__quote">
              <p className="review__text">{review.comment}</p>

              <footer className="review__details">
                <cite className="review__author">{review.user.name}</cite>
                <time className="review__date" dateTime={dateTime}>{date}</time>
              </footer>
            </blockquote>

            <div className="review__rating">{review.rating}</div>
          </div>);
      })
      }
    </div>
  );
};

ReviewsColumn.propTypes = {
  halfReviews: PropTypes.array.isRequired,
};
