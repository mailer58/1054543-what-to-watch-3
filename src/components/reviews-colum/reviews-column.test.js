import React from "react";
import renderer from "react-test-renderer";
import {ReviewsColumn} from "./reviews-column.jsx";

const halfReviews = [{}];

it(`Should ReviewsColumn render correctly`, () => {
  const tree = renderer
    .create(<ReviewsColumn
      halfReviews = {halfReviews} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
