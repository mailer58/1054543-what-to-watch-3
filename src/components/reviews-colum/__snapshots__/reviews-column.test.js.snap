// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ReviewsColumn render correctly 1`] = `
<div
  className="movie-card__reviews-col"
>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      />
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        />
        <time
          className="review__date"
          dateTime="2016-12-20"
        />
      </footer>
    </blockquote>
    <div
      className="review__rating"
    />
  </div>
</div>
`;
